### ISSUE TITLE
BUG: Line breaks are not maintained in chat

### ISSUE THREAD

**Issue by tasdflkjweio** [OvidijusParsiunas/deep-chat#165]
When I type a message into that input that looks like...

```
This is my first line in my first paragraph....and then i start a list:
- An item
- An item
- An item

Then a new paragraph
```
Line breaks are removed when shown in the message area:

```
This is my first line in my first paragraph....and then i start a list:- An item- An item- An itemThen a new paragraph
```

**Comment by OvidijusParsiunas** [comment]
Hi @tasdflkjweio, I will investigate this bug shortly and report to you on the progress.

**Comment by OvidijusParsiunas** [comment]
Hi @tasdflkjweio. This issue has now been fixed in `dev` package versions `9.0.163`.
Please note that the user message is parsed by the [Remarkable](https://jonschlinkert.github.io/remarkable/demo/) library (just like the ai response messages), so the line (-) syntax will be re-formated to be a bullet point, would this be problematic?

Let me know, thanks!

**Comment by tasdflkjweio** [comment]
had a chaotic couple days at work so just checking this out now. 

>  so the line (-) syntax will be re-formated to be a bullet point, would this be problematic?
thanks for asking, it should be fine.

i'll test it out today and let you know soon

**Comment by tasdflkjweio** [comment]
Alright, tested and it's a little better but a bit off still. 

Example 1:
```
line 1
line 2
line 3
line 4
- bullet 1
- bullet 2
line 5
```
is formatted as:
<img width="161" alt="image" src="https://github.com/OvidijusParsiunas/deep-chat/assets/58053652/5bb218e8-7945-40db-8f0c-c56aee299d61">

Example 2:
```
line 1
line 2
line 3

line 4

- bullet 1
- bullet 2


line 5

line 6
line 7
```
is formatted as:
<img width="99" alt="image" src="https://github.com/OvidijusParsiunas/deep-chat/assets/58053652/48628c12-c6f0-4d22-8775-55bbbbd9e9bf">

**Comment by tasdflkjweio** [comment]
Ah i see - it seems Remarkable is expecting CommonMark format. Hrm

**Comment by tasdflkjweio** [comment]
The breaks flag seems to improve this some:
<img width="1039" alt="image" src="https://github.com/OvidijusParsiunas/deep-chat/assets/58053652/2dca904e-80bb-4f19-8c4a-b6c940d06ebb">

That's presumably what you're enabled tho?

**Comment by OvidijusParsiunas** [comment]
Hi @tasdflkjweio.

Deep Chat uses the [Remarkable](https://jonschlinkert.github.io/remarkable/demo/) library's defaults to format text. This way the user will not experience any unexpected behaviour.

To note, if the parent application imports the [highlight.js](https://www.npmjs.com/package/highlight.js?activeTab=readme) module as an [external dependency](https://deepchat.dev/examples/externalModules), we specifically set the configuration to help with coding syntax.

The code can be found [here](https://github.com/OvidijusParsiunas/deep-chat/blob/fae09a367572ac57925f3ef543b81633aaba03dc/component/src/views/chat/messages/remarkable/remarkableConfig.ts#L10).

**Comment by jbrachtie** [comment]
@OvidijusParsiunas When can we expect the next update to the non-dev package? Would love to see this in deep-chat-react (as well as other fixes/improvements I've seen made to the dev package in recent months)!

**Comment by OvidijusParsiunas** [comment]
Hi @jbrachtie
I share your frustration and want to get the big new update out of the way ASAP! I have set myself with a deadline of 11th of July, so if nothing else jumps up in the interim - expect a big update very soon!

**Comment by jbrachtie** [comment]
@OvidijusParsiunas oh wow! Great to know, and certainly understand the effort to get such a big update out. I know a lot has gone into dev since the last release. Thanks for the quick reply and very excited about the next big update :)

**Comment by OvidijusParsiunas** [comment]
Hi folks.
It appears that my previous solution causes strange space issues when rendering code. I have changed the [Remarkable](https://jonschlinkert.github.io/remarkable/demo/) configuration to now always use the `breaks` option mentioned by by @tasdflkjweio.

The fix is available in `deep-chat-dev` and `deep-chat-react-dev` versions `9.0.185`, let me know if this causes issues for anyone. Thanyou!

**Comment by OvidijusParsiunas** [comment]
Hi, everything mentioned in this issue has now been released in Deep Chat version `2.0.0`.
